.clang-tidy

```diff
@@ -0,0 +1,28 @@
+Checks: >
+  -*,
+  bugprone-*
+  misc-*,
+  modernize-*,
+  performance-*,
+  portability-*,
+  readability-*,
+  -fuchsia-trailing-return,
+  -readability-magic-numbers,
+  -modernize-use-nodiscard,
+  -modernize-use-trailing-return-type,
+  -readability-braces-around-statements,
+  -readability-redundant-access-specifiers,
+  -readability-redundant-member-init,
+  -readability-redundant-string-init,
+  -readability-identifier-length,
+CheckOptions:
```

Comment:
I also added naming checks, so we all use the same naming conventions. I chose these based on what we currently use most often.
Let me know if any of these need to change.